plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -142,29 +194,40 @@ func (s *SpanReader) GetTrace(traceID model.TraceID) (*model.Trace, error) {
 	return s.readTrace(dbmodel.TraceIDFromDomain(traceID))
 }
 
-// FindTraces queries for traces.
-func (s *SpanReader) FindTraces(queryParams *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
-	queries, err := BuildQueries(queryParams)
-	if err != nil {
+func validateQuery(p *spanstore.TraceQueryParameters) error {
+	if p == nil {
+		return ErrMalformedRequestObject
+	}
+	if p.ServiceName == "" && len(p.Tags) > 0 {
+		return ErrServiceNameNotSet
+	}
+	if !p.StartTimeMin.IsZero() && !p.StartTimeMax.IsZero() && p.StartTimeMax.Before(p.StartTimeMin) {
```

Comment:
what happens if min or max time are zero?  We should never run queries without these. Either throw an error or use defaults: max = time.Now(), min = max - defaultLookback (like N ours). I think UI always sends these in.